public/app/components/fontViewer/fontViewer.js

```diff
@@ -0,0 +1,477 @@
+angular.module('kibibitCodeEditor')
+
+.directive('kbFontViewer', function() {
+  return {
+    scope: {},
+    bindToController: {
+      openFont: '=kbOpenFont'
+    },
+    controller: 'fontViewerController',
+    controllerAs: 'fontViewerCtrl',
+    templateUrl: 'app/components/fontViewer/fontViewerTemplate.html',
+    link: function(scope, element, attrs, fontViewerCtrl) {
+      scope.$watch('fontViewerCtrl.openFont', function(newOpenFont) {
+        fontViewerCtrl.updateFontView(newOpenFont);
+      });
+      scope.$watch('fontViewerCtrl.previewText', function(textToRender) {
+        if (fontViewerCtrl.font) {
+          fontViewerCtrl.drawPreviewText(textToRender);
+        }
+      });
+    }
+  };
+})
+
+.controller('fontViewerController', [
+  'QuotesService',
+  function(QuotesService) {
+    var vm = this;
+
+    vm.fontSize = 100;
+
+    vm.updateFontView = function(fontObject) {
+      var path = '/api/download/' + encodeURIComponent(fontObject.path);
+      opentype.load(path, function(err, font) {
+        if (err) {
+          alert('Font could not be loaded: ' + err);
+        } else {
+          vm.font = font;
+          onFontLoaded();
+          getQuotes();
+        }
+
+      });
+    };
+
+    function getQuotes() {
```

Comment:
this function is redundent. `QuotesService.getQuotes()` already logs the error if an error is returned.

So I think applying this inline might be easier to understand

```javascript
} else {
  vm.font = font;
  onFontLoaded();
  QuotesService.getQuotes().then(function(quoteList) {
    vm.previewText = quoteList[0];
  });
}
```

If you still want to have a function dedicated to calling the `QuotesService`, give it a name that makes sense for **this controller** and not the `QuotesService`

`QuotesService` gives you quotes, so `QuotesService.getQuotes()` makes sense.

here, `fontViewerController.getQuotes()` doesn't make sense.

First, because it doesn't get quotes. It's just a single quote (since you don't pass any variables to the function, and `QuotesService.getQuotes()` can get a parameter to get more than one quote).
Second, because this function is more about **filling the input field**. So, a name about what this actually does might be more helpful. `function fillInputFieldWithText` or something like that